mid_level_controller:
  # PID gains for thrust, phi, theta
  kp_thrust_x: 1.0
  ki_thrust_x: 0.02
  kd_thrust_x: 0.2

  kp_thrust_y: 1.0
  ki_thrust_y: 0.02
  kd_thrust_y: 0.2

  kp_thrust_z: 4.5
  ki_thrust_z: 0.02
  kd_thrust_z: 2.0

  integral_min: 0.0
  integral_max: 5.0

  min_thrust: 0.0
  max_thrust: 7.848

  dt: 0.01           # Time step for 20 Hz
  
low_level_controller:
  # PID gains for torques
  kp_torque1: 0.02
  ki_torque1: 0.0
  kd_torque1: 0.01

  kp_torque2: 0.02
  ki_torque2: 0.0
  kd_torque2: 0.01

  kp_torque3: 0.02
  ki_torque3: 0.0
  kd_torque3: 0.01

  integral_min: 0.0
  integral_max: 1.0

  dt: 0.004            # Time step for 100 Hz

state_derivative_solver_node:
  # Quadcopter parameters and settings
  mass: 0.450            # Mass of the quadcopter (kg)
  I: [0.00062, 0.0, 0.0, 0.0, 0.00062, 0.0, 0.0, 0.0, 0.00124] # Inertia matrix (flattened row-wise)
  g: [0.0, 0.0, -9.81] # Gravity vector (m/s^2)
  dt: 0.002           # Time step for 500 Hz

  tau_thrust: 0.1
  tau_torque: 0.2

  rho: 1.225
  A_trans: [0.003, 0, 0, 0, 0.003, 0, 0, 0, 0.015]
  Cd_trans: [0.05, 0, 0, 0, 0.05, 0, 0, 0, 0.1]